RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc

```diff
@@ -335,55 +335,28 @@ void ReducedEGProducer::produce(edm::Event& theEvent, const edm::EventSetup& the
       photonPFClusterIsoVals[iid].push_back( (*photonPFClusterIsoHandles[iid])[photonref] );
     }    
     
-    const reco::PhotonCoreRef &photonCore = photon.photonCore();
-    if (!photonCoreMap.count(photonCore)) {
-      photonCores->push_back(*photonCore);
-      photonCoreMap[photonCore] = photonCores->size() - 1;
-    }
+    //link photon core
+    linkPhotonCore(photon, *photonCores, photonCoreMap);
```

Comment:
yes, I can move the call to the photonCore object outside of the function, and just pass the core object.  

I can also add a link/relinkElectronCore method.  let me know.